pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -86,10 +86,12 @@ func (w *PodWatcher) computeChanges(podList []*Pod) ([]*Pod, error) {
 			// We don't check container readiness as init containers are never ready
 			// We check if the container has an ID instead (has run or is running)
 			if !container.IsPending() {
-				if _, found := w.lastSeen[container.ID]; found == false {
+				// We store readiness in the cache key to resubmit the container on pod phase change
+				containerCacheKey := container.ID + "-" + strconv.FormatBool(IsPodReady(pod))
```

Comment:
Where do we evict entries ? Should you isolate the logic to create the key inside a function and reuse for both adding and removing entries ?

Reply:
@ajacquemot the eviction is done purely using the timestamp, we don't recompute keys see https://github.com/DataDog/datadog-agent/blob/40802771ad83b38d2c57bfd9b055dd7a950c0297/pkg/util/kubernetes/kubelet/podwatcher.go#L126-L136
(unit tests were updated for this part also)

Reply:
👍